src/rpc_extra.rs

```diff
@@ -0,0 +1,42 @@
+use anyhow::anyhow;
+use solana_client::rpc_client::RpcClient;
+use solana_sdk::{clock::Epoch, epoch_info::EpochInfo};
+
+/// The number of slots at the beginning of an epoch to look for the first block.
+pub const SLOT_OFFSET: u64 = 40;
+
+/// Applies `f` to the first block in `epoch`.
+pub fn with_first_block<F, A>(
+    client: &RpcClient,
+    epoch: Epoch,
+    epoch_info: &EpochInfo,
+    f: F,
+) -> anyhow::Result<Option<A>>
+where
+    F: Fn(u64) -> anyhow::Result<Option<A>>,
+{
+    // First slot in `epoch`.
+    let first_slot = epoch_info.absolute_slot - epoch_info.slot_index;
```

Comment:
Does this not always return the first slot of the current epoch, regardless of the specified epoch number?
 
```suggestion
    let first_slot = epoch * epoch_info.slots_in_epoch;
```

Reply:
Yes, you are right. I was fixing a bug and introduced a regression.

Reply:
And your solution leads to that bug.